packages/react-client/src/hooks/devices.ts

```diff
@@ -1,33 +1,98 @@
 import { useFishjamContext } from "./fishjamContext";
 import type { Device, AudioDevice } from "../types";
+import { useVideoDeviceManager } from "./deviceManagers/videoDeviceManager";
+import { useAudioDeviceManager } from "./deviceManagers/audioDeviceManager";
+import { useCallback, useEffect, useRef } from "react";
+import { getAvailableMedia, getCorrectedResult } from "../mediaInitializer";
 
 export function useCamera(): Device {
   const {
-    state,
     videoTrackManager: { currentTrack, ...trackManager },
   } = useFishjamContext();
 
+  const { deviceState } = useVideoDeviceManager();
+
   const streamedTrackId = currentTrack?.trackId ?? null;
   const streamedTrack = currentTrack?.track ?? null;
   const stream = currentTrack?.stream ?? null;
-  const devices = state.devices.camera.devices ?? [];
-  const activeDevice = state.devices.camera.deviceInfo;
+  const devices = deviceState.devices ?? [];
+  const activeDevice = deviceState.media?.deviceInfo ?? null;
 
   return { ...trackManager, streamedTrack, streamedTrackId, stream, devices, activeDevice };
 }
 
 export function useMicrophone(): AudioDevice {
   const {
-    state,
     audioTrackManager: { currentTrack, ...trackManager },
   } = useFishjamContext();
 
+  const { deviceState } = useAudioDeviceManager();
+
   const isAudioPlaying = currentTrack?.vadStatus === "speech";
   const streamedTrackId = currentTrack?.trackId ?? null;
   const streamedTrack = currentTrack?.track ?? null;
   const stream = currentTrack?.stream ?? null;
-  const devices = state.devices.microphone.devices ?? [];
-  const activeDevice = state.devices.microphone.deviceInfo;
+  const devices = deviceState.devices ?? [];
+  const activeDevice = deviceState.media?.deviceInfo ?? null;
 
   return { ...trackManager, streamedTrack, streamedTrackId, isAudioPlaying, stream, devices, activeDevice };
 }
+
+type InitializeDevicesProps = { autoInitialize?: boolean };
```

Comment:
Maybe not for this ticket, but... do we need this flag? 🤔

Reply:
I thought we might provide two options to handle this:
1. Doing this automatically - just invoke the hook in the component and the browser will simply ask the user once the component rendered.
2. Doing this manually - the devs might want to invoke the method manually upon handling some action - they could do that by using the method returned by the hook.

I decided the latter one to be the default behavior, nothing happens automatically until verbosely stated.
What do you think?

Reply:
I like @czerwiukk's approach.